services/cd-service/pkg/cmd/server.go

```diff
@@ -62,106 +64,116 @@ func (c *Config) readPgpKeyRing() (openpgp.KeyRing, error) {
 }
 
 func RunServer() {
-	var c Config
-	err := envconfig.Process("kuberpult", &c)
-	if err != nil {
-		log.Fatal(err.Error())
-	}
-
-	pgpKeyRing, err := c.readPgpKeyRing()
-	if err != nil {
-		log.Printf("error reading pgp key ring: %s\n", err)
-		return
-	}
+	logger.Wrap(context.Background(), func(ctx context.Context) error {
+		var c Config
+		err := envconfig.Process("kuberpult", &c)
+		if err != nil {
+			logger.FromContext(ctx).Fatal("config.parse.error", zap.Error(err))
+		}
 
-	if c.ArgoCdPass != "" {
-		_, err := service.ArgocdLogin(c.ArgoCdHost, c.ArgoCdUser, c.ArgoCdPass)
+		pgpKeyRing, err := c.readPgpKeyRing()
 		if err != nil {
-			log.Fatal(err.Error())
+			logger.FromContext(ctx).Fatal("pgp.read.error", zap.Error(err))
 		}
-	}
 
-	ctx := context.Background()
-	repo, err := repository.New(ctx, repository.Config{
-		URL:            c.GitUrl,
-		Path:           "./repository",
-		CommitterEmail: c.GitCommitterEmail,
-		CommitterName:  c.GitCommitterName,
-		Credentials: repository.Credentials{
-			SshKey: c.GitSshKey,
-		},
-		Certificates: repository.Certificates{
-			KnownHostsFile: c.GitSshKnownHosts,
-		},
-		Branch: c.GitBranch,
-		GcFrequency: 20,
-	})
-	if err != nil {
-		log.Fatal(err.Error())
-	}
+		if c.ArgoCdPass != "" {
+			_, err := service.ArgocdLogin(c.ArgoCdHost, c.ArgoCdUser, c.ArgoCdPass)
+			if err != nil {
+				logger.FromContext(ctx).Fatal("argocd.login.error", zap.Error(err))
+			}
+		}
 
-	lockSrv := &service.LockServiceServer{
-		Repository: repo,
-	}
-	deploySrv := &service.DeployServiceServer{
-		Repository: repo,
-	}
+		repo, err := repository.New(ctx, repository.Config{
+			URL:            c.GitUrl,
+			Path:           "./repository",
```

Comment:
I think we should use c.repository here
(I know you didn't introduce it now :) )

Reply:
:+1: let's do that in a seperate MR